### ISSUE TITLE
"The SDK 'Godot.NET.Sdk/4.1.0' specified could not be found."

### ISSUE THREAD

**Issue by LowercaseText** [godotengine/godot#81960]
### Godot version

4.1.0

### System information

Windows 11

### Issue description

I cannot build C# project and I am recieving these two errors
`D:\Dev\FPS\FPS.csproj(0,0): Could not resolve SDK "Godot.NET.Sdk". Exactly one of the probing messages below indicates why we could not resolve the SDK. Investigate and resolve that message to correctly specify the SDK.`

`D:\Dev\FPS\FPS.csproj(0,0): The SDK 'Godot.NET.Sdk/4.1.0' specified could not be found.`

Yes, I have seen other github issues like this one, I have tried to create a nuget.config file in my godot project folder, did not solve the issue.

### Steps to reproduce

This is a semi-fresh install of godot 4.1.0

### Minimal reproduction project

N/A

**Comment by ronyeh** [comment]
I had a similar error when building the Godot Engine from source:


```
HelloWorld.csproj(0,0): Could not resolve SDK "Godot.NET.Sdk". Exactly one of the probing messages below indicates why we could not resolve the SDK. Investigate and resolve that message to correctly specify the SDK.
```

I opened the `*.csproj` file and at the top there's a line that looked like:
```
<Project Sdk="Godot.NET.Sdk/4.2.0-dev">
```

On the nuget page, we see: https://www.nuget.org/packages/Godot.NET.Sdk/

I noticed that there is no such package as 4.2.0-dev, so I changed my csproj file to use 4.2.0.-dev.5 instead:

```
<Project Sdk="Godot.NET.Sdk/4.2.0-dev.5">
```

Now the build works!

In your case, it's kind of weird because 4.1.0 is actually available on nuget. Maybe try changing it to one of the other versions to see if it helps?

I think the Godot C# documentation needs to highlight this roadblock. I imagine it happens once in a while, and it really gives a poor impression to the C# developer who is new to Godot (I imagine there are many of those in recent days).

**Comment by ronyeh** [comment]
I discovered a bug in that when I launch Godot Engine (built from today's head revision), I recreates the `*.csproj` file and overwrites the one that I hand edited.

So even if I changed it to `<Project Sdk="Godot.NET.Sdk/4.2.0-dev.5">` to make it build correctly, Godot will overwrite it with `<Project Sdk="Godot.NET.Sdk/4.2.0-dev">` and my .NET build fails again.

For now, I'll try to keep Godot editor open, but it's annoying that it blows away my csproj file when I launch it.

**Comment by ronyeh** [comment]
@akien-mga I followed those directions and ran the `./modules/mono/build_scripts/build_assemblies.py` with the `--push-nupkgs-local` flag for a local nuget directory. I'm still getting the same error after relaunching Godot.

```
TestProj.csproj(0,0): Could not resolve SDK "Godot.NET.Sdk". Exactly one of the probing messages below indicates why we could not resolve the SDK. Investigate and resolve that message to correctly specify the SDK.
```

I won't file a separate bug since I'm sure it's just a "user error" on my part. I just wish this process were more fool proof. I'm so close.... I'm sure I'll figure it out at some point. :-) I feel like other C# devs will run into this issue at some point too.

**Comment by LowercaseText** [comment]
I think this is an error not so related to godot itself. I tried running godot as an administrator and it was fine (although previously it wasn't working when my normal account was an administrator)
It might just be my computer being a bit pricky because I get a simular error when I try to create a visual studio solution (as a non-administrator) and when I try to install the Godot.NET.Sdk/4.1.0 manually (as a non-adminstrator)
I will post a full solution to this issue when I get it fully figured out and I will close the issue after that.

**Comment by LowercaseText** [comment]
When I was attempting to install the Godot.NET.Sdk/4.1.0 package mannualy using Visual Studio, I was getting an error along the lines of "Access to path C:\Users\georg\.nuget\packages\[something else]"
Turns out there wasn't even a .nuget folder because nuget wasn't allowed to touch it or do anything with it.

In my case, I went to C:\Users\ and right clicked properties>security and then ticked the "Full control" option for "Authenticated users".

sorry if my instructions were a bit unaccurate, I am too lazy to reproduce the issue to tell persisely what errors I was getting.
